SensusService/Protocol.cs

```diff
@@ -1080,7 +1236,17 @@ public void StartWithUserAgreementAsync(string startupMessage, Action callback =
 
                         int consentCodeInt;
                         if (int.TryParse(consentCodeStr, out consentCodeInt) && consentCodeInt == consentCode)
-                            Start();
+                        {
+                            // start protocol or schedule start
+                            if (_startImmediately || (_startTime.Date.Equals(DateTime.Now.Date) && DateTime.Now.TimeOfDay > _startTime.TimeOfDay))
+                                Start();
+                            else
+                                ScheduleStart();
```

Comment:
The above logic needs to be carried out any place we call `Start`. It would be best to put the above in `Start` and move the current `Start` to a private `StartInternal`...or something similar.